From 89972d596cbab4c5289a195bb351529f98be67d3 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <samnazarko@stmteam.com>
Date: Sun, 8 Dec 2013 17:24:29 +0000
Subject: [PATCH] fix issues with iOS 7 AirPlay streaming by changing the
 broadcast name, service API version and reported features to client

---
 xbmc/network/AirPlayServer.cpp  | 4 ++--
 xbmc/network/AirPlayServer.h    | 2 +-
 xbmc/network/AirTunesServer.cpp | 4 ++--
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/xbmc/network/AirPlayServer.cpp b/xbmc/network/AirPlayServer.cpp
index d6979c4..83df5b1 100644
--- a/xbmc/network/AirPlayServer.cpp
+++ b/xbmc/network/AirPlayServer.cpp
@@ -125,9 +125,9 @@
 "<key>deviceid</key>\r\n"\
 "<string>%s</string>\r\n"\
 "<key>features</key>\r\n"\
-"<integer>119</integer>\r\n"\
+"<integer>4294969855</integer>\r\n"\
 "<key>model</key>\r\n"\
-"<string>Xbmc,1</string>\r\n"\
+"<string>AppleTV3,2</string>\r\n"\
 "<key>protovers</key>\r\n"\
 "<string>1.0</string>\r\n"\
 "<key>srcvers</key>\r\n"\
diff --git a/xbmc/network/AirPlayServer.h b/xbmc/network/AirPlayServer.h
index 4c486fb..07a78ea 100644
--- a/xbmc/network/AirPlayServer.h
+++ b/xbmc/network/AirPlayServer.h
@@ -35,7 +35,7 @@
 
 class DllLibPlist;
 
-#define AIRPLAY_SERVER_VERSION_STR "101.28"
+#define AIRPLAY_SERVER_VERSION_STR "160.10"
 
 class CAirPlayServer : public CThread, public ANNOUNCEMENT::IAnnouncer
 {
diff --git a/xbmc/network/AirTunesServer.cpp b/xbmc/network/AirTunesServer.cpp
index 7f40264..199637d 100644
--- a/xbmc/network/AirTunesServer.cpp
+++ b/xbmc/network/AirTunesServer.cpp
@@ -556,9 +556,9 @@ bool CAirTunesServer::StartServer(int port, bool nonlocal, bool usePassword, con
     txt.push_back(std::make_pair("pw",  usePassword?"true":"false"));
     txt.push_back(std::make_pair("vn",  "3"));
     txt.push_back(std::make_pair("da",  "true"));
-    txt.push_back(std::make_pair("vs",  "130.14"));
+    txt.push_back(std::make_pair("vs",  "160.10"));
     txt.push_back(std::make_pair("md",  "0,1,2"));
-    txt.push_back(std::make_pair("am",  "Xbmc,1"));
+    txt.push_back(std::make_pair("am",  "AppleTV3,1"));
 
     CZeroconf::GetInstance()->PublishService("servers.airtunes", "_raop._tcp", appName, port, txt);
   }
-- 
1.8.5.1

